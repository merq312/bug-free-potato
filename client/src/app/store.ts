import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit"
import { logger } from "redux-logger"
import counterReducer from "../features/counter/counterSlice"
import messageReducer from "../features/messages/messageSlice"
import userReducer from "../features/user/userSlice"
import { createSocketMiddleware } from "../features/middleware/socket.middleware"

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    messages: messageReducer,
    user: userReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(createSocketMiddleware(), logger),
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
