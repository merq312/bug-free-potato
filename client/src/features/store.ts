import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit"
import { logger } from "redux-logger"
import counterReducer from "./counter/counterSlice"
import messageReducer from "./message/messageSlice"
import userReducer from "./user/userSlice"
import roomReducer from "./room/roomSlice"
import { createSocketMiddleware } from "./middleware/socket.middleware"

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    messages: messageReducer,
    user: userReducer,
    rooms: roomReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(createSocketMiddleware(), logger),
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
