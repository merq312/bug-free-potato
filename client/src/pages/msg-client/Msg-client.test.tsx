import { render } from '@testing-library/react'
import MsgClientPage from './Msg-client.page'
import { Provider } from 'react-redux'
import { store } from '../../features/store'

describe('Message Client', () => {
  test('renders correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MsgClientPage />
      </Provider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
