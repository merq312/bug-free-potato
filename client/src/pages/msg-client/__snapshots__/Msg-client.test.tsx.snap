// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Client renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col col-start-1 col-end-13 h-full border-l border-r border-gray-300 dark:border-gray-700 overflow-hidden lg:col-start-3 lg:col-end-11"
  >
    <div
      class="h-full flex flex-col grid-rows-tabs lg:grid lg:grid-cols-12 overflow-hidden"
    >
      <div
        class="row-span-full border-r border-b border-gray-300 dark:border-gray-700 lg:col-start-1 lg:col-end-3"
      />
      <div
        class="flex text-gray-800 dark:text-gray-200 mb-1 border-b border-gray-300 dark:border-gray-700 text-base lg:text-xl lg:col-start-3 lg:col-end-13"
      >
        <div
          class="first:border-l px-4 py-2 hover:bg-gray-300 dark:hover:bg-gray-800 border-r border-gray-300 dark:border-gray-700 bg-gray-200 cursor-pointer flex justify-between items-center bg-gray-300 dark:bg-gray-800"
        >
          <span>
            Global
          </span>
        </div>
      </div>
      <div
        class="scroll scrollbar ml-2 border-b border-gray-300 dark:border-gray-700 flex-grow lg:h-auto flex flex-col-reverse lg:col-start-3 lg:col-end-13 overflow-y-scroll"
      >
        <div>
          <div
            class="flex bg-opacity-75 items-center justify-end bg-gray-300 dark:bg-gray-800 mb-2 last:mb-0 lg:last:mb-2 px-4 transition-opacity hover:bg-opacity-40"
          >
            <div
              class="text-gray-800 dark:text-gray-200 flex flex-col justify-start my-1"
            >
              <div
                class="flex justify-end"
              >
                <div
                  class="text-sm mr-1"
                >
                  1 month ago
                </div>
                <div
                  class="text-sm font-bold"
                  data-cy="message-username"
                >
                  Chamila
                </div>
              </div>
              <div
                class="text-right text-lg"
                data-cy="message-content"
              >
                <p>
                  Hello there! Use the input below to send a message.
                </p>
                <p>
                  Click on your name on the bottom left to change your name.
                </p>
                <p>
                  Click on another user's name to open a private chat with them!
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <form
        class="h-full grid lg:grid-rows-1 grid-cols-12 items-center justify-around"
      >
        <input
          autocomplete="off"
          class="text-base h-full border-r border-gray-300 dark:border-gray-700 lg:text-2xl col-start-1 col-end-13 lg:col-start-1 lg:col-end-3 px-3 text-gray-800 dark:text-gray-200 bg-gray-200 dark:bg-gray-900 focus:outline-none resize-none"
          data-cy="input-username"
          name="user-name"
          type="text"
          value="Guest"
        />
        <div
          class="col-start-1 lg:col-start-3 col-end-13 flex bg-gray-300 dark:bg-gray-800"
        >
          <textarea
            autocomplete="off"
            class="no-scrollbar flex-grow min-w-0 my-3 px-3 py-1.5 bg-gray-300 dark:bg-gray-800 text-gray-800 dark:text-gray-200 focus:outline-none resize-none"
            data-cy="input-message"
            name="user-input"
            rows="1"
          />
          <button
            class="px-2 sm:px-4 text-gray-800 dark:text-gray-200 font-bold"
            data-cy="send-message"
            type="submit"
          >
            Send
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-paper-plane fa-w-16 ml-2 h-6"
              data-icon="paper-plane"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
